src/main/java/software/amazon/awssdk/crt/io/EventLoopGroup.java

```diff
@@ -58,6 +58,58 @@ private void onCleanupComplete() {
 
     public CompletableFuture<Void> getShutdownCompleteFuture() { return shutdownComplete; }
 
+
+    /*
+     * Static interface for access to a default, lazily-created event loop group for users who don't
+     * want to deal with the associated resource management.  Client bootstraps will use this event loop
+     * group if they are passed a null value.
+     */
+
+    /**
+     * Sets the number of threads for the static default event loop group, should it ever be created.  Has no
+     * effect if the static default event loop group has already been created.
+     *
+     * @param numThreads number of threads for the static default event loop group
+     */
+    public static void setStaticDefaultNumThreads(int numThreads) {
+        synchronized (EventLoopGroup.class) {
+            staticDefaultNumThreads = numThreads;
+        }
+    }
+
+    /**
+     * Closes the static default event loop group, if it exists.  Primarily intended for tests that use the static
+     * default event loop group, before they call waitForNoResources().
+     */
+    public static void closeStaticDefault() {
+        synchronized (EventLoopGroup.class) {
+            if (staticDefaultEventLoopGroup != null) {
+                staticDefaultEventLoopGroup.close();
+            }
+            staticDefaultEventLoopGroup = null;
+        }
+    }
+
+    /**
+     * Gets the static default event loop group, creating it if necessary
+     * @return the static default event loop group
+     */
+    static EventLoopGroup getOrCreateStaticDefault() {
+        EventLoopGroup elg = null;
+        synchronized (EventLoopGroup.class) {
+            if (staticDefaultEventLoopGroup == null) {
+                staticDefaultEventLoopGroup = new EventLoopGroup(staticDefaultNumThreads);
+            }
+
+            elg = staticDefaultEventLoopGroup;
+        }
+
+        return elg;
+    }
+
+    private static int staticDefaultNumThreads = 1;
```

Comment:
I feel like I would be more surprised to see 12-16 threads spring up out of nowhere when I write "client.builder().create()" then seeing 1 thread spring up, but that's just me.  What happens with the comparable netty event loop group?

Reply:
Netty uses 2 * cores

https://github.com/netty/netty/blob/b908c82b7b60afae16f510f6cbb5093b9a69ee33/transport/src/main/java/io/netty/channel/MultithreadEventLoopGroup.java#L40

It might be an issue if customers are using multiple SDK clients because it seems the default number of threads can't be changed if it's already created.

Reply:
Hmm, seems reasonable to follow their lead then.